pkg/controllers/jobset_controller.go

```diff
@@ -731,7 +733,10 @@ func labelAndAnnotateObject(obj metav1.Object, js *jobset.JobSet, rjob *jobset.R
 	jobName := placement.GenJobName(js.Name, rjob.Name, jobIdx)
 
 	// Set labels on the object.
-	labels := collections.CloneMap(obj.GetLabels())
+	labels := maps.Clone(obj.GetLabels())
```

Comment:
Why do we need to clone the map here? similar question to the annotations clone below

Reply:
That is a good question. 
TBH, I have the same question since we already set the constructed labels and annotations to object the below: https://github.com/kubernetes-sigs/jobset/blob/4b1eddd94402905260342791be255044d58d8a55/pkg/controllers/jobset_controller.go#L786-L787

I thought this redundant clone came from historical reasons.
Anyway, let me remove these clones.

Reply:
Done.
If you want to keep these clones, let me know.